src/MapView/MapView.tsx

```diff
@@ -1,37 +1,35 @@
-import {forwardRef, useEffect, useImperativeHandle, useMemo, useRef} from 'react';
-import Mapbox, {MarkerView} from '@rnmapbox/maps';
+import {forwardRef, useEffect, useImperativeHandle, useRef, useState} from 'react';
+import Mapbox, {MapState, MarkerView} from '@rnmapbox/maps';
 import {View} from 'react-native';
 import {MapViewProps, MapViewHandle} from './MapViewTypes';
 import Direction from './Direction';
 import Utils from './utils';
 
 const MapView = forwardRef<MapViewHandle, MapViewProps>(function MapView(
-    {accessToken, style, styleURL, pitchEnabled, mapPadding, initialState, waypoints, directionCoordinates, directionStyle},
+    {accessToken, style, styleURL, pitchEnabled, mapPadding, initialState, waypoints, directionCoordinates, directionStyle, isFocused},
     ref,
 ) {
     const cameraRef = useRef<Mapbox.Camera>(null);
+    const [isIdle, setIsIdle] = useState(false);
 
-    const bounds = useMemo(() => {
-        if (!waypoints || waypoints.length === 0) {
-            return undefined;
-        }
+    useEffect(() => {
+        if (isFocused) return;
+        setIsIdle(false);
+    }, [isFocused]);
+
+    useEffect(() => {
+        if (!waypoints || !waypoints.length || !isIdle || !isFocused) return;
 
         if (waypoints.length === 1) {
-            cameraRef.current?.flyTo(waypoints[0].coordinate);
```

Comment:
we don't have "flying" behavior anymore?

Reply:
Instead of `flyTo` method we are using `setCamera`

Reply:
does it still animates the map?

Reply:
@gegham-khachatryan I need a confirmation on this point. 

@allroundexperts can you test this behavior?

Reply:
@hayata-suenaga Absolutely, `flyTo` method can also be used to animate a map. The second argument, animationDuration, determines the duration of the animation. My apologies for the delayed response. If you have any further questions or concerns, please don't hesitate to ask.

Reply:
you mean the second argument to `setCamera` right? that's good let's use it (and pass some numbers to animate the camera)

Reply:
Both methods (`flyTo` and `setCamera`) supports animation. In addition of changing the center `setCamera` can control zoom also. `animationDuration` for `flyTo` is second argument, for `setCamera` is an options property.